### PR TITLE
Removed dependency of openSSL for authentication with .p12 keys

### PR DESCRIPTION
**Pull request by googlyrahman**
Test Results = https://paste.googleplex.com/6510605917224960
Removed dependency of openSSL for authentication with .p12 keys

### CODE REVIEW

File: gcs_oauth2_boto_plugin/test_oauth2_client.py [hunk 1]
```diff
@@ -174,6 +174,34 @@ def _RunGetAccessTokenTest(self, expected_rapt=None):
         token_3.expiry)
     self.assertEqual(token_3.rapt_token, expected_rapt)
 
+class TestCreateP12ServiceAccountCredentials(unittest.TestCase):
+  
+  @mock.patch.object(oauth2_client.PKCS12Signer, 'from_string')
```
Comment (shilpi23pandey) [inline gcs_oauth2_boto_plugin/test_oauth2_client.py:179]: This patcher is provided by the unittest mock which is available in py 3.3+. For the earlier versions,  It relies on mock third party lib. Will this compile for the earlier versions? https://github.com/GoogleCloudPlatform/gcs-oauth2-boto-plugin/blob/master/gcs_oauth2_boto_plugin/test_oauth2_client.py#L33

Comment (googlyrahman) [reply gcs_oauth2_boto_plugin/test_oauth2_client.py:179]: It looks like before python version 3.3 we're importing mock in our tests, and it's same mock which is part of unit-test in python version > 3.3

```
mock is now part of the Python standard library, available as [unittest.mock](https://docs.python.org/dev/library/unittest.mock.html) in Python 3.3 onwards.
```

Comment (shilpi23pandey) [reply gcs_oauth2_boto_plugin/test_oauth2_client.py:179]: Got it! thanks.

**Review by shilpi23pandey — APPROVED** [review]

**Review by sushantkhanna — APPROVED** [review]
Shilpi has already approved this change. Going ahead and approving + merging the change as Shilpi does not have access yet.

**Pull request merged**